Classes/Domain/Site/SiteHashService.php

```diff
@@ -115,7 +116,8 @@ protected function getDomainByPageIdAndReplaceMarkers($pageId, $allowedSitesConf
      */
     protected function getAvailableSites()
     {
-        return Site::getAvailableSites();
+        $siteRepository = GeneralUtility::makeInstance(SiteRepository::class);
```

Comment:
I would propose to make the siteRepository a constructor dependecy and use it directly. I think getAvailableSites abd getSiteByPageId have only been introduced as proctected methods because the old static way Site::getAvailableSites and Site::getSiteByPageId was not mockable in the test (because of the static).